widgetssdk/src/main/java/com/glia/widgets/internal/notification/device/NotificationManager.kt

```diff
@@ -3,59 +3,64 @@ package com.glia.widgets.internal.notification.device
-            createCallChannel()
-            createScreenSharingChannel()
-        }
+    init {
+        createNotificationChannels()
     }
 
-    @RequiresApi(api = Build.VERSION_CODES.O)
-    private fun createCallChannel() {
-        if (notificationManager.getNotificationChannel(NotificationFactory.NOTIFICATION_CALL_CHANNEL_ID) == null) {
-            /* Screen sharing notification importance should have the possible highest value,
-            because it shows the ongoing audio/video streaming and it is preferable to be as high as possible in notifications list */
-            val importance = NotificationManager.IMPORTANCE_HIGH
-            val notificationChannel = NotificationChannel(
-                NotificationFactory.NOTIFICATION_CALL_CHANNEL_ID,
-                applicationContext.getString(R.string.android_notification_audio_call_channel_name),
-                importance
-            )
-            notificationManager.createNotificationChannel(notificationChannel)
-        }
+    private fun createNotificationChannels() {
+        createCallChannel()
+        createScreenSharingChannel()
     }
 
-    @RequiresApi(api = Build.VERSION_CODES.O)
-    private fun createScreenSharingChannel() {
-        if (notificationManager.getNotificationChannel(NotificationFactory.NOTIFICATION_SCREEN_SHARING_CHANNEL_ID) == null) {
-            /* Screen sharing notification importance should have the possible highest value,
+    private fun createCallChannel() = createChannelIfDoesNotExist(
+        NotificationFactory.NOTIFICATION_CALL_CHANNEL_ID,
+        R.string.android_notification_audio_call_channel_name,
+        /* Call notification importance should have the possible highest value,
+        because it shows the ongoing audio/video streaming and it is preferable to be as high as possible in notifications list */
+        NotificationManager.IMPORTANCE_HIGH
+    )
+
+    private fun createScreenSharingChannel() = createChannelIfDoesNotExist(
+        NotificationFactory.NOTIFICATION_SCREEN_SHARING_CHANNEL_ID,
+        R.string.android_notification_screen_sharing_channel_name,
+        /* Screen sharing notification importance should have the possible highest value,
             because it affects the Media projection service initialization and streaming timing */
-            val importance = NotificationManager.IMPORTANCE_HIGH
-            val notificationChannel = NotificationChannel(
-                NotificationFactory.NOTIFICATION_SCREEN_SHARING_CHANNEL_ID,
-                applicationContext.getString(R.string.android_notification_screen_sharing_channel_name),
-                importance
+        NotificationManager.IMPORTANCE_HIGH
+    )
+
+    override fun createSecureMessagingChannel() = createChannelIfDoesNotExist(
+        NotificationFactory.NOTIFICATION_SECURE_MESSAGING_CHANNEL_ID,
+        R.string.android_notification_secure_messaging_channel_name,
+        NotificationManager.IMPORTANCE_HIGH
+    )
```

Comment:
Shouldn't it go to the application default channel instead of the one defined by our SDK?

Reply:
Unfortunately, there is no way to use application default channel for push notifications, it is reserved and used only by the system. So, I'm rolling back to the approach with creating channel for SC pushes.